modules/network/multivpc/main.tf

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+locals {
+  # this input variable is validated to be in CIDR format
+  network_name          = coalesce(replace(var.network_prefix, "_", "-"), replace(var.deployment_name, "_", "-"))
+  global_ip_cidr_suffix = split("/", var.global_ip_address_range)[1]
+  network_new_bits      = var.network_cidr_suffix - local.global_ip_cidr_suffix
+  subnetwork_bits       = ceil(log(var.network_count, 2))
+  additional_networks = [
+    for vpc in module.vpcs :
+    merge(var.network_template, { subnetwork = vpc.subnetwork_name, subnetwork_project = var.project_id })
+  ]
+}
+
+resource "terraform_data" "vpc_validation" {
+  lifecycle {
```

Comment:
Sorry, I missed this. You should define `input = var.subnetwork_cidr_suffix`

and then use (at line 21 above)

```
  subnetwork_new_bits      = terraform_data. vpc_validation.output - local.global_ip_cidr_suffix
```

You might rename `vpc_validation` to `subnetwork_cidr_suffix` for some clarity.

Reply:
Did you mean renaming `network_cidr_suffix`?

Reply:
Yes, in both instances to subnetwork_cidr_suffix